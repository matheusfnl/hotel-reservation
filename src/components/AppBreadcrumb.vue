<template>
  <q-breadcrumbs class="text-brown">
    <template v-slot:separator>
      <q-icon size="1.5em" name="chevron_right" color="primary" />
    </template>

    <q-breadcrumbs-el
      v-for="route in breadcrumbItems"
      :key="route"
      :label="$t(`routes.${route}`)"
    />
  </q-breadcrumbs>
</template>

<script lang="ts" setup>
defineProps<{
  breadcrumbItems: string[]
}>()
</script>
