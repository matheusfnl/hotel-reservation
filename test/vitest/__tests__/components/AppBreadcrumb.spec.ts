import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { QBreadcrumbsEl } from 'quasar'

import AppBreadcrumb from '@/components/AppBreadcrumb.vue'
import type { BreadcrumbItem } from '@/types'

const mountWrapper = (props: { breadcrumbItems: Array<string | BreadcrumbItem> }) => {
  return mount(AppBreadcrumb, {
    props,
  })
}

describe('AppBreadcrumb', () => {
  it('should render breadcrumb with simple routes', () => {
    const breadcrumbItems = ['home', 'hotels']
    const wrapper = mountWrapper({ breadcrumbItems })
    const items = wrapper.findAllComponents(QBreadcrumbsEl)

    expect(wrapper.html()).toMatchSnapshot()
    expect(items.length).toBe(2)
    expect(items[0]!.props('label')).toBe(wrapper.vm.$t('routes.home'))
    expect(items[1]!.props('label')).toBe(wrapper.vm.$t('routes.hotels'))
  })

  it('should render breadcrumb with complex routes', () => {
    const breadcrumbItems = [
      'home',
      { label: 'destination.name', options: { name: 'Hotel', local: 'BH' } },
    ]
    const wrapper = mountWrapper({ breadcrumbItems })
    const items = wrapper.findAllComponents(QBreadcrumbsEl)

    expect(wrapper.html()).toMatchSnapshot()
    expect(items.length).toBe(2)
    expect(items[1]!.props('label')).toBe('"Hotel" em BH')
    expect(items[1]!.props('to')).toEqual({ name: 'destination.name' })
  })
})
