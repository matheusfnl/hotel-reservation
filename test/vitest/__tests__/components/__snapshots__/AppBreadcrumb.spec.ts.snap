// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppBreadcrumb > should render breadcrumb with complex routes 1`] = `
"<div class="q-breadcrumbs text-brown">
  <div class="flex items-center justify-start q-gutter-sm">
    <div class="flex items-center text-primary"><span class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable">Início</span></div>
    <div class="q-breadcrumbs__separator"><i class="q-icon text-primary notranslate material-icons" style="font-size: 1.5em;" aria-hidden="true" role="presentation">chevron_right</i></div>
    <div class="flex items-center q-breadcrumbs--last"><span class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable">"Hotel" em BH</span></div>
  </div>
</div>"
`;

exports[`AppBreadcrumb > should render breadcrumb with simple routes 1`] = `
"<div class="q-breadcrumbs text-brown">
  <div class="flex items-center justify-start q-gutter-sm">
    <div class="flex items-center text-primary"><span class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable">Início</span></div>
    <div class="q-breadcrumbs__separator"><i class="q-icon text-primary notranslate material-icons" style="font-size: 1.5em;" aria-hidden="true" role="presentation">chevron_right</i></div>
    <div class="flex items-center q-breadcrumbs--last"><span class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable">Hotéis</span></div>
  </div>
</div>"
`;
